### ISSUE TITLE
Search field autocomplete results are not immediately apparent to TalkBack users [Android]

### ISSUE THREAD

**Issue by jonnyjames** [covid-alert-ny/covid-green-app#301]
Autocomplete suggestions are shown after the search field found on the “Select your county” overlays, with all possible options shown when nothing is entered in the search field. The message “No results found for this search. Try with something else” is shown when there are no results. This message and the fact that there are autocomplete results below the search field is not immediately apparent to TalkBack users.

![Search - autocomlete results](https://user-images.githubusercontent.com/7034772/92133903-ea6cbb00-edd6-11ea-86e0-b7034734fcde.png)

**Remediation Guidance**

When there are results shown (initially and when results are found that match the user inputted data) append something like “Showing autocomplete suggestions” to the `accessibilityLabel`.
```javascript
<TextInput accessibilityLabel="Search, Showing autocomplete suggestions" />
When there are no results, append the “No results …” message to the accessibilityLabel. The visible message text can be then hidden from TalkBack users, as the text will now be redundant.

<TextInput accessibilityLabel="Search, No results found for this search. Try with something else" />

<View accessibilityElementsHidden={true}>
  <Text>No results found for this search. Try with something else</Text>
</View>
```

**Comment by floridemai** [comment]
@jonnyjames This should be addressed in the next build. Not that the search label doesn't update right after you make edits, but the labels and roles should be ok.

**Comment by jonnyjames** [comment]
@floridemai The autocomplete results (or lack of) are not announced by TalkBack. I am not hearing the suggested "Search, Showing autocomplete suggestions" or "Search, No results found for this search. Try with something else". 

Another approach, is to use a live region for Android. Here is the documentation:

[https://reactnative.dev/docs/accessibility#accessibilityliveregion](https://reactnative.dev/docs/accessibility#accessibilityliveregion)

**Comment by AlanSl** [comment]
Done in https://github.com/project-vagabond/covid-green-app/pull/391

This is a tricky one because we have to work around an android-specific bug in React Native's TextInput itself - https://github.com/facebook/react-native/issues/26739.

I tried a few approaches and this seemed to work best:

 - If it's Android and the screen reader is on, the text input does not have placeholder text. This stops that bug from preventing the initial accessibility label from being read. It says "Search" on initial focus, and whether or not there are autocomplete suggestions available, as well as the usual system search box speech.
   - To ensure partially-sighted screenreader users can still press the search box the usual way, if the placeholder text is not shown the text input is made wider and given an outline
 - On entering text, on Android, the accessibility label is updated and includes the full current text value and whether there are suggestions or not. This way a user knows if their input has caused there to no longer be any suggestions available and if so what the full text that caused this is.

Please retest and let me know if this is satisfactory - keeping in mind that we're working around a bug in React Native itself so we're a little limited in what we can do.

**Comment by helloantoine** [comment]
Tested. Looks good to me. Thx Alan!
